src/tla/__rewire_finite_sets_ext_in_apalache.tla

```diff
@@ -0,0 +1,166 @@
+ * define the Apalache-compatible behavior.
+ *
+ * These definitions are automatically rewired by the Apalache importer.
+ *
+ * Compare with the original definitions in FiniteSetsExt.tla:
+ *
+ * https://github.com/tlaplus/CommunityModules/blob/master/modules/FiniteSetsExt.tla
+ *)
+
+LOCAL INSTANCE Integers
+LOCAL INSTANCE FiniteSets
+LOCAL INSTANCE __apalache_folds 
+LOCAL INSTANCE Functions
+
+(**
+ * Fold op over the elements of set using base as starting value.
+ * Note that this FoldSet is different from Apalache!FoldSet.
+ *
+ * Example:
+ *         FoldSet(LAMBA x,y : x + y, 0, 0 .. 10) = 55
+ *
+ * @type: ((a, b) => b, b, Set(a)) => b;
+ *)
+FoldSet(op(_, _), base, set) ==
+    \* ApalacheFoldSet is accumulating the result in the left argument,
+    \* whereas FinitSetsExt!FoldSet is accumulating the result
+    \* in the right argument.
+    LET OpSwap(x, y) == op(y, x) IN
+    __ApalacheFoldSet(OpSwap, base, set)
+
+(**
+ * Calculate the sum of the elements in set.
+ *
+ * Example: Sum(0 .. 10) = 55
+ *
+ * @type: Set(Int) => Int;
+ *)
+SumSet(set) ==
+   LET Plus(x, y) == x + y IN
+   __ApalacheFoldSet(Plus, 0, set)
+
+(**
+ * Calculuate the product of the elements in set.
+ *
+ * Example: Product(1 .. 3) = 6
+ *
+ * @type: Set(Int) => Int;
+ *)
+ProductSet(set) ==
+   LET Mult(x, y) == x * y IN
+   __ApalacheFoldSet(Mult, 1, set)
+
+(**
+ * An alias for FoldSet. ReduceSet was used instead of FoldSet in
+ * earlier versions of the community modules.
+ *
+ * @type: ((a, b) => b, Set(a), b) => b;
+ *)
+ReduceSet(op(_, _), set, acc) == 
+   FoldSet(op, set, acc)
+
+(**
+ * Starting from base, apply op to f(x), for all x \in S, in an arbitrary
+ * order. It is assumed that op is associative and commutative.
+ *
+ * Example: FlattenSet({{1},{2}}) = {1,2}
+ *
+ * @type: Set(Set(a)) => Set(a);
+ *)
+FlattenSet(S) ==
+  LET SetUnion(A, B) == A \union B IN
+  __ApalacheFoldSet(SetUnion, {}, S)
+
+(**
+ * The symmetric difference of two sets.
+ *
+ * The symmetric difference of sets A and B is the set containing all
+ * elements that are present in either A or B but not in their intersection.
+ *
+ * @type: (Set(a), Set(a)) => Set(a);
+ *)
+SymDiff(A, B) ==
+    { x \in A: x \notin B } \union { x \in B: x \notin A }
+
+(**
+ * Quantify the elements in S matching the predicate (LAMDBA) P.
+ *
+ * This operator is overridden by FiniteSetsExt#quantify whose
+ * implementation does *not* enumerate the intermediate set! This is
+ * the only advantage that Quantify(...) has over Cardinality(...).
+ *
+ * Example: Quantify(1..9, LAMBDA n : n % 2 = 0) = 4
+ *
+ * @type: (Set(a), (a => Bool)) => Int;
+ *)
+Quantify(S, P(_)) ==
+   Cardinality({s \in S : P(s)})
+
+
+(**
+ * A k-subset ks of a set S has Cardinality(ks) = k.  The number of
+ * k-subsets of a set S with Cardinality(S) = n is given by the binomial
+ * coefficients n over k.  A set S with Cardinality(S) = n has 2^n
+ * k-subsets.  \A k \notin 0..Cardinality(S): kSubset(k, S) = {}.
```

Comment:
Can you add a one-line summary of what this operator actually does? (return the k-subsets of S)

Reply:
This is just a copy of the comments from the community modules.

Reply:
Since we can't sync with them, we might as well diverge?